crates/signer/src/manager/dirk.rs

```diff
@@ -0,0 +1,917 @@
+use std::{
+    collections::{HashMap, HashSet},
+    fmt::Debug,
+    fs,
+    path::PathBuf,
+};
+
+use alloy::{hex, primitives::FixedBytes, rpc::types::beacon::constants::BLS_SIGNATURE_BYTES_LEN};
+use blsful::inner_types::{Field, G2Affine, G2Projective, Group, Scalar};
+use cb_common::{
+    commit::request::{ConsensusProxyMap, ProxyDelegation, SignedProxyDelegation},
+    config::DirkConfig,
+    constants::COMMIT_BOOST_DOMAIN,
+    signature::compute_domain,
+    signer::{BlsPublicKey, BlsSignature, ProxyStore},
+    types::{Chain, ModuleId},
+};
+use rand::Rng;
+use tonic::transport::{Channel, ClientTlsConfig};
+use tracing::{info, trace, warn};
+use tree_hash::TreeHash;
+
+use crate::{
+    error::SignerModuleError::{self, DirkCommunicationError},
+    proto::v1::{
+        account_manager_client::AccountManagerClient, lister_client::ListerClient,
+        sign_request::Id as SignerId, signer_client::SignerClient, Account as DirkAccount,
+        DistributedAccount, GenerateRequest, ListAccountsRequest, ResponseState, SignRequest,
+        UnlockAccountRequest,
+    },
+};
+
+#[derive(Clone, Debug)]
+enum WalletType {
+    Simple,
+    Distributed,
+}
+
+#[derive(Clone, Debug)]
+struct HostInfo {
+    domain: String,
```

Comment:
I think `server_name` would be better to avoid confusion with signing domains

Reply:
Yes good idea, done in 27eb6831420584e2739bfc83e33dfb8ada4aefc7